### ISSUE TITLE
Share GPU assumptions

### ISSUE THREAD

**Issue by sradc** [microsoft/visual-chatgpt#41]
Hey, it looks like [this](https://github.com/microsoft/visual-chatgpt/blob/e2c74844b3555741e6d47fb76cdda6c54319f989/visual_chatgpt.py#L803) assumes that there are 8 GPUs available. Are you able to provide a bit more info about that? (I.e. what GPUs do you run this on, and recommend running on?)

(Maybe worth adding some info on this in the readme?)

**Comment by b2zer** [comment]
Anything goes, so long as you have 70 GB of VRAM... lol

I found this fork to be super useful (which removes a lot of the models that would otherwise just give you an OOM - but at least you can use *some* while chatting to ChatGPT and having it create images for you!):

https://github.com/rupeshs/visual-chatgpt/tree/add-colab-support

It's good for toying around with this proof-of-concept, but you either need to pay for some cloud compute deluxe or be living in a small server room to enjoy the whole thing. 

Good thing they said there will be an API soon "in a few days". Together with GPT-4 rumors turning into "next week", I guess I'll settle playing with what fits in my VRAM and then try to get my hands on the API. :-)

**Comment by sradc** [comment]
I did get this running, on 8x NVIDIA A100 40 Gb, in the end, but various bugs prevented it from fully working (for one, the masking for inpainting wasn't working, not sure if the fork uses/fixes this model?).

> Anything goes, so long as you have 70 GB of VRAM...

Not quite, I tried running it on 8x NVIDIA Tesla V100, 16 Gb. But got OOM on _one of the cards_ when trying to generate an image. I.e. the cards need to be big enough to run the models allocated to them.

...Looking forward to the multimodal APIs coming soon, as you say.
